From 6bee95d3339b5213a5869532bca8b2314dd92887 Mon Sep 17 00:00:00 2001
From: "Andrew F. Davis" <afd@ti.com>
Date: Sat, 2 Apr 2016 01:06:22 +0000
Subject: [PATCH 45/46] HACK: ARM: dts: am437x-idk-evm: Add non-caching OCMC
 RAM node

Split the OCMC RAM node making the upper-half a non-caching section.
This prevents remote processors from accessing stale data.

Real fix will involve ensuring we flush the cache before signaling
the read from any external devices.

Signed-off-by: Andrew F. Davis <afd@ti.com>
Tested-by: Jason Reeder <jreeder@ti.com>
---
 arch/arm/boot/dts/am4372.dtsi | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/am4372.dtsi b/arch/arm/boot/dts/am4372.dtsi
index 05c5037..ed7fa6a 100644
--- a/arch/arm/boot/dts/am4372.dtsi
+++ b/arch/arm/boot/dts/am4372.dtsi
@@ -1146,10 +1146,15 @@
 
 		ocmcram: ocmcram@40300000 {
 			compatible = "mmio-sram";
-			reg = <0x40300000 0x40000>; /* 256k */
+			reg = <0x40300000 0x10000>; /* 64k */
 			map-exec;
 		};
 
+		ocmcram_nocache: ocmcram_nocache@40310000 {
+			compatible = "mmio-sram";
+			reg = <0x40310000 0x30000>; /* 192k */
+		};
+
 		dcan0: can@481cc000 {
 			compatible = "ti,am4372-d_can", "ti,am3352-d_can";
 			ti,hwmods = "d_can0";
-- 
1.9.1

